target/linux/realtek/image/rtl838x.mk

```diff
@@ -148,6 +148,17 @@ define Device/panasonic_m8eg-pn28080k
 endef
 TARGET_DEVICES += panasonic_m8eg-pn28080k
 
+define Device/tplink_sg2008p-v1
+  SOC := rtl8380
+  IMAGE_SIZE := 6144k
```

Comment:
I think this will also generate errors on sysupgrade files larger than 6 MiB. Have you built any images larger than this limit already? If not, you could add some blob of random data to `files/` to blow up the image size.

Reply:
I found that the `-initramfs` is marginally larger, so it will trigger an error first. I didn't bother with size-checking the other images, but since you called my bluff...

Reply:
Okay, final fix is to use KERNEL_SIZE and IMAGE_SIZE.